resteasy-jaxrs/src/main/java/org/jboss/resteasy/core/MediaTypeMap.java

```diff
@@ -272,7 +272,7 @@ private CachedMediaTypeAndClass(Class clazz, MediaType mediaType)
       {
          this.clazz = new WeakReference(clazz);
          this.mediaType = mediaType;
-         int result = getClazz().hashCode();
+         int result = clazz.hashCode();

```

Comment:
Totally agree but what I am saying is that the `clazz` I'm using in `int result = clazz.hashCode();` refers to `Class clazz` argument and not `WeakReference<Class<?>> clazz` property. In other words:
`int result = clazz.hashCode()` is not the same as `int result = **this**.clazz.hashCode()` where `this.clazz` refers to `WeakReference<Class<?>> clazz` in this case.

I should have renamed the variable to be clearer like this:

```
...
      // we need a weak reference because of possible hot deployment
      // Although, these reference should get cleared up with any add() invocation
private WeakReference<Class<?>> clazzReference;

private CachedMediaTypeAndClass(Class clazz, MediaType mediaType){
         this.clazzReference= new WeakReference(clazz);
         this.mediaType = mediaType;
         int result = clazz.hashCode();
     ...
}
```